drivers/timer/Kconfig

```diff
@@ -30,12 +30,12 @@ config APIC_TIMER_IRQ_PRIORITY
 	  This option specifies the IRQ priority used by the local APIC timer.
 
 config APIC_TIMER_TSC
-	bool "Use invariant TSC for z_timer_cycle_get_32()"
+	bool "Use invariant TSC for clock_cycle_get_32()"
 	help
 	  If your CPU supports invariant TSC, and you know the ratio of the
 	  TSC frequency to CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC (the local APIC
 	  timer frequency), then enable this for a much faster and more
-	  accurate z_timer_cycle_get_32().
+	  accurate clock_cycle_get_32().
```

Comment:
I agree with @dcpleung that it would be best to more clearly identify which clock is involved. 

`z_timer_cycle_get_32()` is aliased by `arch_k_cycle_get_32()` which in turn is aliased by `k_cycle_get_32()`.

The rate it advances is provided by `sys_clock_hw_cycles_per_sec()` and some implementations also use `sys_clock_` as a standard prefix in the system clock driver(s) for the architecture.

So it would seem to make sense to have this be `sys_clock_get_32()`, or `sys_clock_cycle_get_32()`.